podpac/core/node.py

```diff
@@ -969,7 +972,7 @@ def wrapper(self, coordinates, output=None):
         key = cache_key
         cache_coordinates = coordinates.transpose(*sorted(coordinates.dims))  # order agnostic caching
 
-        if not self.cache_update and self.has_cache(key, cache_coordinates):
+        if not self.cache_update and self.cache_output and self.has_cache(key, cache_coordinates):
```

Comment:
@jmilloy What do you think about this addition. Make sense? I adjusted the documentation to reflect this. 

Basically, if a user realizes they have cache_output = True and then turn it off, the would STILL get cached results WITHOUT this addition.

Reply:
Makes sense.

Another option would be to observe `cache_output` and remove the output from the cache when it is changed to False. Without that, a user could turn cache_output off, then turn it back on and get the old results again.

BTW, I think `cache_update` is confusing. I think you should consider renaming it something like `force_eval` or `force_reevaluate` or `force_ignore_cached_output`.

Reply:
Do you want to set cache_dataset in LoadFileMixin to False by default, also?

Either way, we should fix the has_cache check there in the same way:

```diff
diff --git a/podpac/core/data/file_source.py b/podpac/core/data/file_source.py
index 2749da1..2f38932 100644
--- a/podpac/core/data/file_source.py
+++ b/podpac/core/data/file_source.py
@@ -88,7 +88,7 @@ class LoadFileMixin(S3Mixin):
     @cached_property
     def dataset(self):
         # use the _dataset_caching_node "stub" here because the only node attr we care about is the source
-        if self._dataset_caching_node.has_cache(key="dataset"):
+        if self.cache_dataset and self._dataset_caching_node.has_cache(key="dataset"):
             data = self._dataset_caching_node.get_cache(key="dataset")
             with BytesIO(data) as f:
                 return self._open(BytesIO(data), cache=False)
```

Reply:
why is dataset even using the has_cache? don't have you have to set `use_cache_ctrl` on the `cached_property` decorator to use the cache?

Reply:
> BTW, I think `cache_update` is confusing. I think you should consider renaming it something like `force_eval` or `force_reevaluate` or `force_ignore_cached_output`.

I agree completely. I keep having a hard time finding this setting. The idea was that all the cache-related attributes would start with `cache`... but I usually end uplooking for `force_eval`.

Reply:
This one is a special case because it has to cache the binary representation instead of the object itself. So it does the has_cache and put_cache itself. I think it would fail to cache the object in some cases if you use `use_cache_ctrl=True` directly.

Actually, it is now *also* special now because I am using a stub node (with only the source attribute but no data_key or outputs) to do the caching of the dataset.

Reply:
`cache_force_eval`?

Reply:
> Another option would be to observe `cache_output` and remove the output from the cache when it is changed to False. Without that, a user could turn cache_output off, then turn it back on and get the old results again.

i think in this case the user should set `update_cache` (or `force_eval`). I wouldn't want to remove something from the cache without the user's permission.

Reply:
> i think in this case the user should set update_cache (or force_eval). I wouldn't want to remove something from the cache without the user's permission.

okay, makes sense

Reply:
then again, by that reasoning, you could remove `cache_output` altogether, and just use `force_eval` from the get-go...

Reply:
> Another option would be to observe `cache_output` and remove the output from the cache when it is changed to False. Without that, a user could turn cache_output off, then turn it back on and get the old results again.

i think in this case the user should set `update_cache` (or `force_eval`). I wouldn't want to remove something from the cache without the user's permission. 




> 
> 
> > why is dataset even using the has_cache?
> 
> This one is a special case because it has to cache the binary representation instead of the object itself. So it does the has_cache and put_cache itself. I think it would fail to cache the object in some cases if you use `use_cache_ctrl=True` directly.
> 
> Actually, it is now _also_ special now because I am using a stub node (with only the source attribute but no data_key or outputs) to do the caching of the dataset.

Got it. I missed that the `put_cache` was in the `open` method. In that case yes, I think it should be `False` by default. In fact, I believe I've run into this problem where I get the cached dataset instead of the source dataset (which has changed during the session and I'm opening it again to check it!). I ended up just restarting my kernel -- thought it was Zarr caching issue...

Thanks for pointing these out @jmilloy

Reply:
> 
> 
> then again, by that reasoning, you could remove `cache_output` altogether, and just use `force_eval` from the get-go...

Let's see, right now `force_eval=True` will ignore the cache, but NOT update the output unless  `cache_output=True`. I think that's useful in case you want to do an experiment without losing what you have in the cache... subtle though... I'll have to think about that.